### PR TITLE
[DOC] Doc for field processing

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Documents command-line options `-n`, `-p`, `-a`, and `-F`, each affecting field processing.

I've put the new file into directory `doc/command_line/` (instead of `doc/`) because I'm planning more files documenting more CLI options.  These may eventually be gathered into an inclusive `doc/command_line_options.md`.

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
Converted to Draft (temporarily) -- there's a little more that can be done here.

**Comment by BurdetteLamar** [comment]
Ready for review (option `-l` added).

File: doc/command_line/field_processing.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+```
+while gets
+  # Your Ruby code.
+end
+```
+
+Note that `gets` reads the next line and sets global variable `$_`
+to the last read line;
+note also that character `'$'` must be escaped as `'\$'`:
```
Comment (nobu) [inline doc/command_line/field_processing.md:42]: That escape is needed when enclosed in double quotes or no quotes.
Is it for the sake of Windows, not to use single quotes?

File: doc/command_line/field_processing.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+Option `-F`, when given with option `-a`,
+specifies that its argument is to be the input field separator to be used for splitting:
+
+```sh
+$ ruby -an -Fs -e "p \$F" desiderata.txt
+["Go placidly amid the noi", "e and the ha", "te,\n"]
+["and remember what peace there may be in ", "ilence.\n"]
+["A", " far a", " po", "", "ible, without ", "urrender,\n"]
+["be on good term", " with all per", "on", ".\n"]
+```
```
Comment (nobu) [inline doc/command_line/field_processing.md:94]: I think it's worth mentioning that the arguments are compiled as regular expressions.

```sh
$ ruby -an -F'[.,]\s*' -e 'p $F' desiderata.txt
["Go placidly amid the noise and the haste"]
["and remember what peace there may be in silence"]
["As far as possible", "without surrender"]
["be on good terms with all persons"]
```

File: doc/command_line/field_processing.md (hunk 1 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md]: Fixed (7 places).

File: doc/command_line/field_processing.md (hunk 2 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md]: Note added.

**Comment by BurdetteLamar** [comment]
Changed heading levels (downward), with a view to possible eventual file inclusion.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, is this ready to merge, do you think?

File: doc/command_line/field_processing.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+```
+text = <<EOT
+Go placidly amid the noise and the haste,
+and remember what peace there may be in silence.
+As far as possible, without surrender,
+be on good terms with all persons.
+EOT
+File.write('desiderata.txt', text)
+```
```
Comment (peterzhu2118) [inline doc/command_line/field_processing.md:21]: IMO this is more confusing than it has to. We can just say that `desiderata.txt` contains the contents.

File: doc/command_line/field_processing.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+```sh
+$ ruby -an -e 'p $F' desiderata.txt
+["Go", "placidly", "amid", "the", "noise", "and", "the", "haste,"]
+["and", "remember", "what", "peace", "there", "may", "be", "in", "silence."]
+["As", "far", "as", "possible,", "without", "surrender,"]
+["be", "on", "good", "terms", "with", "all", "persons."]
+```
+
+The default field separator used for the splitting is the input field separator `$;`.
```
Comment (peterzhu2118) [inline doc/command_line/field_processing.md:80]: Do we want to mention that the default field separator used for splitting the newline is `$/`?

File: doc/command_line/field_processing.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+```sh
+$ ruby -an -Fs -e 'p $F' desiderata.txt
+["Go placidly amid the noi", "e and the ha", "te,\n"]
+["and remember what peace there may be in ", "ilence.\n"]
+["A", " far a", " po", "", "ible, without ", "urrender,\n"]
+["be on good term", " with all per", "on", ".\n"]
+```
+
+The argument may be a regular expression;
+see String#split.
```
Comment (peterzhu2118) [inline doc/command_line/field_processing.md:96]: I'm not sure how to specify a regular expression here. I tried `-F/the/` or `-F\s` and none of them worked. Can you explain?

File: doc/command_line/field_processing.md (hunk 3 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md:21]: As in some of the other docs, this gives the user a cut-and-paste route to actually running the examples.  But I'll replace with a remark if you prefer.

Comment (peterzhu2118) [reply doc/command_line/field_processing.md:21]: Isn't this more work though? The user has to copy and paste this script and run this script. If we directly have the script, then user can just copy and paste the text.

**Comment by peterzhu2118** [comment]
Can you also clarify how this doc will be used? For example, where will it be linked from. Right now, it's not very easy to find by the average reader.

File: doc/command_line/field_processing.md (hunk 5 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md:96]: I was trying to accommodate the comment at https://github.com/ruby/ruby/pull/10008#discussion_r1493670245.  I'm not sure exactly what we should say.

File: doc/command_line/field_processing.md (hunk 3 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md]: FIxed.

**Comment by BurdetteLamar** [comment]
> Can you also clarify how this doc will be used? For example, where will it be linked from. Right now, it's not very easy to find by the average reader.

I'm working toward docs in `doc/command_line`, that will cover all the command-line options, likely included into `doc/command_line/options.md`.

I'd like then to add a URL address to the help text, referring the reader to these robust write-ups.  (Too bad it can't be a link -- or can it?)

File: doc/command_line/field_processing.md (hunk 5 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md:95]: I'll have an update for this soon.

File: doc/command_line/field_processing.md (hunk 4 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md]: Done.

File: doc/command_line/field_processing.md (hunk 5 above)
Comment (BurdetteLamar) [reply doc/command_line/field_processing.md]: Fixed (added @nobu example and remark).

**Pull request merged**